sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_base.py

```diff
@@ -0,0 +1,213 @@
+
+        :keyword Source source: Required. The source.
+        :keyword JWTTokenAuth auth: Required.
+        :keyword int idle_timeout: Required.
+        :keyword network_trace: Required.
+        :keyword retry_policy: Required.
+        :keyword str client_name: Required.
+        :keyword dict link_properties: Required.
+        :keyword properties: Required.
+        :keyword link_credit: Required. The prefetch.
+        :keyword keep_alive_interval: Required. Missing in pyamqp.
+        :keyword desired_capabilities: Required.
+        :keyword streaming_receive: Required.
+        :keyword message_received_callback: Required.
+        :keyword timeout: Required.
+        """
+    @abstractmethod
+    def open_receive_client(self, *, handler, client, auth):
+        """
+        Opens the receive client.
+        :param ReceiveClient handler: The receive client.
+        :param ~azure.eventhub.EventHubConsumerClient client: The consumer client.
+        """
+
+    @abstractmethod
+    def create_token_auth(self, auth_uri, get_token, token_type, config, **kwargs):
+        """
+        Creates the JWTTokenAuth.
+        :param str auth_uri: The auth uri to pass to JWTTokenAuth.
+        :param get_token: The callback function used for getting and refreshing
+         tokens. It should return a valid jwt token each time it is called.
+        :param bytes token_type: Token type.
+        :param ~azure.eventhub._configuration.Configuration config: EH config.
+
+        :keyword bool update_token: Whether to update token. If not updating token,
+         then pass 300 to refresh_window. Only used by uamqp.
+        """
+
+    @abstractmethod
+    def create_mgmt_client(self, address, mgmt_auth, config):
+        """
+        Creates and returns the mgmt AMQP client.
+        :param _Address address: Required. The Address.
+        :param JWTTokenAuth mgmt_auth: Auth for client.
+        :param ~azure.eventhub._configuration.Configuration config: The configuration.
+        """
+
+    @abstractmethod
+    def get_updated_token(self, mgmt_auth):
+        """
+        Return updated auth token.
+        :param mgmt_auth: Auth.
+        """
+
+    @abstractmethod
+    def mgmt_client_request(self, mgmt_client, mgmt_msg, **kwargs):
+        """
+        Send mgmt request.
+        :param AMQP Client mgmt_client: Client to send request with.
+        :param str mgmt_msg: Message.
+        :keyword bytes operation: Operation.
+        :keyword operation_type: Op type.
+        :keyword status_code_field: mgmt status code.
+        :keyword description_fields: mgmt status desc.
+        """
+
+    @abstractmethod
+    def get_error(self, error, message, *, condition=None):
+        """
+        Gets error and passes in error message, and, if applicable, condition.
+        :param error: The error to raise.
+        :param str message: Error message.
+        :param condition: Optional error condition. Will not be used by uamqp.
```

Comment:
leave as param with default value, or keyword? -- to keep consistent you made another uamqp-only a keyword arg

Reply:
oh must've missed the inconsistency, good catch!